File: EpicerieOnline2/Core/Models/Order.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace EpicerieOnline2.Core.Models
{
    [Table("Orders")]
    public class Order
    {
        public int Id { get; set; }

        public int CustomerId { get; set; }

        public Customer Customer { get; set; }

        public DateTime Date { get; set; }

        public string Name { get; set; }

        [Required]
        [StringLength(255)]
        public string Address { get; set; }

        public double PriceTotal { get; set; }

        public bool IsValidate { get; set; }

        public bool IsCompleted { get; set; }

        public ICollection<OrderProduct> Products { get; set; }

        public Order()
        {
            Products = new Collection<OrderProduct>();
        }



    }
}

File: ClientApp/src/app/models/order.ts
import { Product } from "./product";


export interface SaveForOrderProduct {
  id: number;
  quantity?: number;
}


export interface Order {
  id: number;
  customerId: number;
  name: string;
  address: string;
  priceTotal: number;
  products: SaveForOrderProduct[];
}

export interface CustomerFromServer {
  id: number;
  email: string;
  password?: string;
  roleId: number;
}


export interface OrderFromServer {
  id: number;
  customer: CustomerFromServer;
  date: string;
  isCompleted: boolean;
  isValidate: boolean;
  name: string;
  address: string;
  priceTotal: number;
  products: Product[];
}


export interface UpdateOrder {
  id: number;
  customerId: number;
  name: string;
  address: string;
  priceTotal: number;
  isCompleted: boolean;
  isValidate: boolean;

  products: UpdateOrderProduct[];
}


export interface UpdateOrderProduct {
  id: number;
  quantity?: number;
  title?: string;
}
